rasa/core/brokers/kafka.py

```diff
@@ -124,67 +124,73 @@ def publish(
                     f"Could not publish message to kafka url '{self.url}'. "
                     f"Failed with error: {e}"
                 )
-                connected = self.producer.bootstrap_connected()
-                if not connected:
-                    self._close()
+                try:
+                    self.producer.flush(timeout=1)
```

Comment:
mhm why are we flushing here? previous behaviour was to try to receonnect - flushing will wait for all pending messages to be produced, not sure if that helps here. does confluent kafka internally try to reconnect on connection failure?

Reply:
> does confluent kafka internally try to reconnect on connection failure?

Not too sure since I couldn't find anything in their docs (the python client API docs) about reconnection. I'll check the way connection is verified in analytics and replace accordingly, as using `flush` was not a very bright idea 😄

Reply:
Addressed: https://github.com/RasaHQ/rasa/pull/11765/commits/ad960d46c115ca8433779f2ab8e539d2449cac92